crates/hdk_derive/src/lib.rs

```diff
@@ -44,16 +43,22 @@ impl Parse for EntryDef {
         for var in vars {
             if let Some(segment) = var.path.segments.first() {
                 match segment.ident.to_string().as_str() {
-                    "id" => match var.lit {
-                        syn::Lit::Str(s) => {
+                    "id" => match var.value {
+                        syn::Expr::Lit(syn::ExprLit {
+                            lit: syn::Lit::Str(s),
+                            ..
+                        }) => {
```

Comment:
var does not have a `lit` field anymore. Instead its a more generic `value` field that needs to be matched to a lit expression.